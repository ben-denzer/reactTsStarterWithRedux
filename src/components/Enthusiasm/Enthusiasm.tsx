import * as React from 'react';
import { EnthusiasmWrapper } from './EnthusiasmStyles';
import getExclamationMarks from '../../utils/getExclamationMarks';

export interface Props {
  name: string;
  enthusiasmLevel?: number;
  onIncrement: () => void;
  onDecrement: () => void;
  onReset: () => void;
}

const Enthusiasm = (props: Props): JSX.Element => {
  const {
    name,
    enthusiasmLevel = 1,
    onDecrement,
    onIncrement,
    onReset
  } = props;
  if (!enthusiasmLevel || enthusiasmLevel < 1) {
    return (
      <div>
        <div>You Broke The System</div>{' '}
        <button id="reset" onClick={onReset}>
          {' '}
          Reset{' '}
        </button>
      </div>
    );
  }
  return (
    <EnthusiasmWrapper>
      <div className="greeting">
        Hello {name + getExclamationMarks(enthusiasmLevel)}
      </div>
      <div>
        <button id="decrement" onClick={onDecrement}>
          {' '}
          -{' '}
        </button>
        <button id="increment" onClick={onIncrement}>
          {' '}
          +{' '}
        </button>
        <button id="reset" onClick={onReset}>
          {' '}
          Reset{' '}
        </button>
      </div>
    </EnthusiasmWrapper>
  );
};

export default Enthusiasm;
