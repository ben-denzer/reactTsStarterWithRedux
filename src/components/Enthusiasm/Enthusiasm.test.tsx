import * as React from 'react';
import { shallow } from 'enzyme';
import Enthusiasm, { Props } from './Enthusiasm';

describe('Enthusiasm', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      name: 'Nobody',
      enthusiasmLevel: 2,
      onIncrement: jest.fn(),
      onDecrement: jest.fn(),
      onReset: jest.fn()
    };
  });
  it('should render without crashing', () => {
    shallow(<Enthusiasm {...props} />);
  });
  it('should call onDecrement on button click', () => {
    const wrapper = shallow(<Enthusiasm {...props} />);
    wrapper.find('#decrement').simulate('click');
    expect(props.onDecrement).toHaveBeenCalledTimes(1);
  });
  it('should call onIncrement on button click', () => {
    const wrapper = shallow(<Enthusiasm {...props} />);
    wrapper.find('#increment').simulate('click');
    expect(props.onIncrement).toHaveBeenCalledTimes(1);
  });
  it('should call onReset on button click', () => {
    const wrapper = shallow(<Enthusiasm {...props} />);
    wrapper.find('#reset').simulate('click');
    expect(props.onReset).toHaveBeenCalledTimes(1);
  });
  it('should match snapshot with no enthusiasmLevel', () => {
    const testProps = { ...props, enthusiasmLevel: undefined };
    const wrapper = shallow(<Enthusiasm {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot with enthusiasmLevel', () => {
    const wrapper = shallow(<Enthusiasm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot with invalid enthusiasmLevel', () => {
    const testProps = { ...props, enthusiasmLevel: -1 };
    const wrapper = shallow(<Enthusiasm {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
