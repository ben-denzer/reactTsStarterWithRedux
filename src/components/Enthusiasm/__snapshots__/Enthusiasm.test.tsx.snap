// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enthusiasm should match snapshot with enthusiasmLevel 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Enthusiasm
    enthusiasmLevel={2}
    name="Nobody"
    onDecrement={[MockFunction]}
    onIncrement={[MockFunction]}
    onReset={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="greeting"
        >
          Hello 
          Nobody!!
        </div>,
        <div>
          <button
            id="decrement"
            onClick={[MockFunction]}
          >
             
            -
             
          </button>
          <button
            id="increment"
            onClick={[MockFunction]}
          >
             
            +
             
          </button>
          <button
            id="reset"
            onClick={[MockFunction]}
          >
             
            Reset
             
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            "Nobody!!",
          ],
          "className": "greeting",
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          "Nobody!!",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              id="decrement"
              onClick={[MockFunction]}
            >
               
              -
               
            </button>,
            <button
              id="increment"
              onClick={[MockFunction]}
            >
               
              +
               
            </button>,
            <button
              id="reset"
              onClick={[MockFunction]}
            >
               
              Reset
               
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "-",
                " ",
              ],
              "id": "decrement",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "-",
              " ",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "+",
                " ",
              ],
              "id": "increment",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "+",
              " ",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Reset",
                " ",
              ],
              "id": "reset",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Reset",
              " ",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="greeting"
          >
            Hello 
            Nobody!!
          </div>,
          <div>
            <button
              id="decrement"
              onClick={[MockFunction]}
            >
               
              -
               
            </button>
            <button
              id="increment"
              onClick={[MockFunction]}
            >
               
              +
               
            </button>
            <button
              id="reset"
              onClick={[MockFunction]}
            >
               
              Reset
               
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello ",
              "Nobody!!",
            ],
            "className": "greeting",
          },
          "ref": null,
          "rendered": Array [
            "Hello ",
            "Nobody!!",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                id="decrement"
                onClick={[MockFunction]}
              >
                 
                -
                 
              </button>,
              <button
                id="increment"
                onClick={[MockFunction]}
              >
                 
                +
                 
              </button>,
              <button
                id="reset"
                onClick={[MockFunction]}
              >
                 
                Reset
                 
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "-",
                  " ",
                ],
                "id": "decrement",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "-",
                " ",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "+",
                  " ",
                ],
                "id": "increment",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "+",
                " ",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Reset",
                  " ",
                ],
                "id": "reset",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Reset",
                " ",
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Enthusiasm should match snapshot with invalid enthusiasmLevel 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Enthusiasm
    enthusiasmLevel={-1}
    name="Nobody"
    onDecrement={[MockFunction]}
    onIncrement={[MockFunction]}
    onReset={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          You Broke The System
        </div>,
        " ",
        <button
          id="reset"
          onClick={[MockFunction]}
        >
           
          Reset
           
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You Broke The System",
        },
        "ref": null,
        "rendered": "You Broke The System",
        "type": "div",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Reset",
            " ",
          ],
          "id": "reset",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Reset",
          " ",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            You Broke The System
          </div>,
          " ",
          <button
            id="reset"
            onClick={[MockFunction]}
          >
             
            Reset
             
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You Broke The System",
          },
          "ref": null,
          "rendered": "You Broke The System",
          "type": "div",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Reset",
              " ",
            ],
            "id": "reset",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Reset",
            " ",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Enthusiasm should match snapshot with no enthusiasmLevel 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Enthusiasm
    enthusiasmLevel={undefined}
    name="Nobody"
    onDecrement={[MockFunction]}
    onIncrement={[MockFunction]}
    onReset={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="greeting"
        >
          Hello 
          Nobody!
        </div>,
        <div>
          <button
            id="decrement"
            onClick={[MockFunction]}
          >
             
            -
             
          </button>
          <button
            id="increment"
            onClick={[MockFunction]}
          >
             
            +
             
          </button>
          <button
            id="reset"
            onClick={[MockFunction]}
          >
             
            Reset
             
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            "Nobody!",
          ],
          "className": "greeting",
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          "Nobody!",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              id="decrement"
              onClick={[MockFunction]}
            >
               
              -
               
            </button>,
            <button
              id="increment"
              onClick={[MockFunction]}
            >
               
              +
               
            </button>,
            <button
              id="reset"
              onClick={[MockFunction]}
            >
               
              Reset
               
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "-",
                " ",
              ],
              "id": "decrement",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "-",
              " ",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "+",
                " ",
              ],
              "id": "increment",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "+",
              " ",
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Reset",
                " ",
              ],
              "id": "reset",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Reset",
              " ",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="greeting"
          >
            Hello 
            Nobody!
          </div>,
          <div>
            <button
              id="decrement"
              onClick={[MockFunction]}
            >
               
              -
               
            </button>
            <button
              id="increment"
              onClick={[MockFunction]}
            >
               
              +
               
            </button>
            <button
              id="reset"
              onClick={[MockFunction]}
            >
               
              Reset
               
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello ",
              "Nobody!",
            ],
            "className": "greeting",
          },
          "ref": null,
          "rendered": Array [
            "Hello ",
            "Nobody!",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                id="decrement"
                onClick={[MockFunction]}
              >
                 
                -
                 
              </button>,
              <button
                id="increment"
                onClick={[MockFunction]}
              >
                 
                +
                 
              </button>,
              <button
                id="reset"
                onClick={[MockFunction]}
              >
                 
                Reset
                 
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "-",
                  " ",
                ],
                "id": "decrement",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "-",
                " ",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "+",
                  " ",
                ],
                "id": "increment",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "+",
                " ",
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Reset",
                  " ",
                ],
                "id": "reset",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Reset",
                " ",
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
